src/runtime/auto-prompt-manager.ts

```diff
@@ -200,18 +201,23 @@ export class AutoPromptManager {
       };
     }
 
-    const shouldShowAutoPrompt = await this.shouldShowAutoPrompt_(
-      clientConfig,
-      entitlements,
-      params.autoPromptType
-    );
+    const shouldShowMonetizationPrompt =
+      this.shouldShowMonetizationPromptFromClientConfig(
+        clientConfig,
+        params.autoPromptType
+      ) &&
+      !entitlements.enablesThis() &&
```

Comment:
I'm wondering if we can check entitlement at the top of `showAutoPrompt_` and exit the flow if there's a valid entitlement?

Reply:
Moved to the top and no tests fail. Also removed a redundant entitlements check from `shouldShowBlockingPrompt`. Which, do we want to rename to `canDisplayBlockingPrompt`?